### ISSUE TITLE
Error on MacOS - checking AWS STS access – cannot get role ARN for current session: NoCredentialProviders: no valid providers in chain

### ISSUE THREAD

**Issue by nimbusscale** [weaveworks/eksctl#321]
**What happened?**
When running `eksctl get cluster` on my MacBook pro I get the following error message:

```
2018-11-20T10:53:43-05:00 [✖]  checking AWS STS access – cannot get role ARN for current session: NoCredentialProviders: no valid providers in chain
caused by: EnvAccessKeyNotFound: failed to find credentials in the environment.
SharedCredsLoad: failed to load profile, .
EC2RoleRequestError: no EC2 instance role found
caused by: RequestError: send request failed
caused by: Get http://169.254.169.254/latest/meta-data/iam/security-credentials: dial tcp 169.254.169.254:80: connect: host is down
```

**What you expected to happen?**
A list of the one cluster created by eksctl previously running in us-west-2.

**How to reproduce it?**
`eksctl create cluster` or `eksctl get cluster` on MacOS.

**Anything else we need to know?**
aws creds file (cross account role is being used):
```
$ cat ~/.aws/credentials
[bct-master-long-term]
aws_access_key_id = SNIP
aws_secret_access_key = <SNIP>
region = us-west-2

[default]
role_arn = arn:aws:iam::899826514230:role/SandboxAdmin
source_profile = bct-master-long-term
region = us-west-2
```

aws config file:
```
$ cat ~/.aws/config 
[default]
region = us-west-2
```

The following AWS CLI commands work just fine:
```
$ aws eks list-clusters
{
    "clusters": [
        "adorable-party-1541899386"
    ]
}

$ aws sts get-caller-identity 
{
    "Account": "899826514230", 
    "UserId": "AROAIK3I7IPINEIATJ2DK:botocore-session-1542727810", 
    "Arn": "arn:aws:sts::899826514230:assumed-role/SandboxAdmin/botocore-session-1542727810"
}

$ aws sts assume-role --role-arn "arn:aws:iam::899826514230:role/SandboxAdmin" --role-session-name "RoleSessionTest"
{
    "AssumedRoleUser": {
        "AssumedRoleId": "AROAIK3I7IPINEIATJ2DK:RoleSessionTest", 
        "Arn": "arn:aws:sts::899826514230:assumed-role/SandboxAdmin/RoleSessionTest"
    }, 
    "Credentials": {
        "SecretAccessKey": <SNIP>, 
        "SessionToken": <SNIP>,
        "Expiration": "2018-11-20T16:55:30Z", 
        "AccessKeyId": <SNIP>
    }
}
```
eksctl works just fine on Linux for me (using same creds and config file from above). From what I can see on the Internet this could be related to the Go SDK. I tried various troubleshooting items to see if I can get it work, but no luck. All produce the same error.

```
$ export AWS_PROFILE=default
$ eksctl get cluster
```

```
$ export AWS_REGION=us-west-2
$ eksctl get cluster
```
```
$ eksctl get cluster --region=us-west-2
```

```
$ export AWS_SDK_LOAD_CONFIG=1
$ eksctl get cluster
$ eksctl get cluster --profile=us-west-2
```
When trying to create a cluster:
```
$ eksctl create cluster
2018-11-20T11:44:34-05:00 [ℹ]  using region us-west-2
2018-11-20T11:44:45-05:00 [✖]  checking AWS STS access – cannot get role ARN for current session: NoCredentialProviders: no valid providers in chain
caused by: EnvAccessKeyNotFound: failed to find credentials in the environment.
SharedCredsLoad: failed to load profile, .
EC2RoleRequestError: no EC2 instance role found
caused by: RequestError: send request failed
caused by: Get http://169.254.169.254/latest/meta-data/iam/security-credentials: dial tcp 169.254.169.254:80: connect: host is down
```

**Versions**
```
$ aws --version
aws-cli/1.16.58 Python/2.7.10 Darwin/16.7.0 botocore/1.12.48
$ uname -a
Darwin Joes-MBP.home 16.7.0 Darwin Kernel Version 16.7.0: Thu Jun 15 17:36:27 PDT 2017; root:xnu-3789.70.16~2/RELEASE_X86_64 x86_64
$ eksctl version
2018-11-20T11:17:49-05:00 [ℹ]  versionInfo = map[string]string{"builtAt":"2018-11-09T16:15:40Z", "gitCommit":"191474b2b0a6e6856b5f9c652c38b5f2f01bf7c9", "gitTag":"0.1.11"}
```

aws-iam-authenticator downloaded from url below, not sure how to get it to print a version string).
```
https://amazon-eks.s3-us-west-2.amazonaws.com/1.10.3/2018-07-26/bin/darwin/amd64/aws-iam-authenticator`
```

**Logs**
```
$ eksctl get cluster -v4
2018-11-20T11:39:21-05:00 [✖]  checking AWS STS access – cannot get role ARN for current session: NoCredentialProviders: no valid providers in chain
caused by: EnvAccessKeyNotFound: failed to find credentials in the environment.
SharedCredsLoad: failed to load profile, .
EC2RoleRequestError: no EC2 instance role found
caused by: RequestError: send request failed
caused by: Get http://169.254.169.254/latest/meta-data/iam/security-credentials: dial tcp 169.254.169.254:80: connect: host is down
```

**Comment by errordeveloper** [comment]
Thanks for detailed report! I am on holiday at the moment, but @christopherhein maybe able to help here.

**Comment by nimbusscale** [comment]
I am using the `default` profile and I was able to figure it out.

For some reason on Mac I needed to set the AWS_SHARED_CREDENTIALS_FILE Env Var, even though I'm using the default path. 

```
$ eksctl get clusters
2018-12-04T10:45:14-08:00 [✖]  checking AWS STS access – cannot get role ARN for current session: NoCredentialProviders: no valid providers in chain
caused by: EnvAccessKeyNotFound: failed to find credentials in the environment.
SharedCredsLoad: failed to load profile, .
EC2RoleRequestError: no EC2 instance role found
caused by: RequestError: send request failed
caused by: Get http://169.254.169.254/latest/meta-data/iam/security-credentials: dial tcp 169.254.169.254:80: connect: host is down
$ export AWS_SHARED_CREDENTIALS_FILE=~/.aws/credentials
$ eksctl get clusters
NAME				REGION
attractive-sheepdog-1543796753	us-west-2
```

I don't need to set this on Linux.  I assume this is a bug with the Go SDK and not an Eksctl bug.

**Comment by nimbusscale** [comment]
Closing this issue with the assumption this is not an eksctl bug.

**Comment by TreverW** [comment]
I still see this issue on Mac OSX 10.14.4 but it is intermittent.  Sometimes it works, sometimes it doesn't.  

I am using: 
eksctl version 0.1.29, 
aws cli version aws-cli/1.16.140 Python/3.7.3 Darwin/18.5.0 botocore/1.12.130

I have tried setting the AWS_SHARED_CREDENTIALS_FILE environment variable as suggested by @nimbusscale.  

Here is an example session where it worked then didn't.

```bash
# aws cli commands work fine (consistently)
$ aws eks list-clusters
{
    "clusters": [
        "kubernetes-cd5"
    ]
}

# Intermittently, eksctl commands fail.
$ eksctl create nodegroup --config-file=cluster-cd5.yml --only nodes-test
[ℹ]  using region us-east-1
[✖]  checking AWS STS access – cannot get role ARN for current session: RequestError: send request failed
caused by: Post https://sts.amazonaws.com/: net/http: TLS handshake timeout

$ export AWS_SHARED_CREDENTIALS_FILE=/Users/twilhelm/.aws/credentials

# Setting the env var seemed to have helped, at least for one command...
$ eksctl create nodegroup --config-file=cluster-cd5.yml --only nodes-test
[ℹ]  using region us-east-1
[ℹ]  nodegroup "nodes-test" will use "ami-0bd7eb0889ba1a301" [AmazonLinux2/1.11]
[!]  retryable error (Throttling: Rate exceeded
	status code: 400, request id: 44311633-606d-11e9-b4e7-236a52c9e7fc) from cloudformation/DescribeStacks - will retry after delay of 731ms
[!]  retryable error (Throttling: Rate exceeded
	status code: 400, request id: 44ccb9c0-606d-11e9-b69a-1967de8b00be) from cloudformation/DescribeStacks - will retry after delay of 1.594s
[ℹ]  3 nodegroup(s) were filtered out: --only="nodes-test" was given, 5 nodegroup(s) (cluster-management1, ignite, nodes-default2) already exist
[ℹ]  will create a CloudFormation stack for each of 1 nodegroups in cluster "kubernetes-cd5"
[ℹ]  1 task: { create nodegroup "nodes-test" }
[ℹ]  buildings nodegroup stack "eksctl-kubernetes-cd5-nodegroup-nodes-test"
[ℹ]  --nodes-min=6 was set automatically for nodegroup nodes-test
[ℹ]  --nodes-max=6 was set automatically for nodegroup nodes-test
[ℹ]  deploying stack "eksctl-kubernetes-cd5-nodegroup-nodes-test"
[ℹ]  adding role "nodes-kubernetes-cd5" to auth ConfigMap
[ℹ]  nodegroup "nodes-test" has 0 node(s)
[ℹ]  waiting for at least 6 node(s) to become ready in "nodes-test"
[ℹ]  nodegroup "nodes-test" has 6 node(s)
[ℹ]  node "ip-172-22-104-124.ec2.internal" is ready
[ℹ]  node "ip-172-22-112-83.ec2.internal" is ready
[ℹ]  node "ip-172-22-134-152.ec2.internal" is ready
[ℹ]  node "ip-172-22-156-21.ec2.internal" is ready
[ℹ]  node "ip-172-22-166-71.ec2.internal" is ready
[ℹ]  node "ip-172-22-179-29.ec2.internal" is ready
[✔]  created 1 nodegroup(s) in cluster "kubernetes-cd5"
[ℹ]  checking security group configuration for all nodegroups
[ℹ]  all nodegroups have up-to-date configuration

# next command failed
$ eksctl get ng --cluster=kubernetes-cd5 --output=json
[✖]  checking AWS STS access – cannot get role ARN for current session: RequestError: send request failed
caused by: Post https://sts.amazonaws.com/: net/http: TLS handshake timeout
```

**Comment by TreverW** [comment]
Is it possible to reopen this issue?  I can try to help provide any details necessary to reproduce and fix it.

Thanks.

**Comment by Davidrjx** [comment]
same error on debian distribution. executing "eksctl create ..." outputs error resembling
```
checking AWS STS access – cannot get role ARN for current session: InvalidClientTokenId: The security token included in the request is invalid.
        status code: 403, request id: 9a89b74d-807b-11e9-9198-3b3470540988
```

**Comment by MitchyBAwesome** [comment]
Also seeing this issue on **macOS 10.13.6** with eksctl version **0.1.32**. As above, all other aws cli commands are working as expected. 

```
Checking AWS STS access – cannot get role ARN for current session: RequestError: send request failed
caused by: Post https://sts.amazonaws.com/: net/http: TLS handshake timeout
```

**Comment by benhg** [comment]
I'm still seeing this issue as well, on MacOS 10.14

**Comment by errordeveloper** [comment]
@benhg from our slack conversation, I was under impression you resolved this on your end and it was a local configuration issue, correct?

**Comment by benhg** [comment]
Yes, sorry to bother you guys.

**Comment by nimbusscale** [comment]
I just reopened as I saw there were a few others who had the issue. I've since moved on to work on other things. Please feel free to reclose if you don't think this is an issue that needs to be looked at.

I won't reopen again.

**Comment by errordeveloper** [comment]
Thanks everyone!

**Comment by imago-storm** [comment]
I see this as well, on MacOS 10.14, constantly. I've tried setting AWS_SHARED_CREDENTIALS_FILE variable. My eksctl version is [ℹ]  version.Info{BuiltAt:"", GitCommit:"", GitTag:"0.3.0"}.

**Comment by MLDMoritz** [comment]
Experiencing this on Ubuntu.

**Comment by mirr254** [comment]
I am also experiencing this on Ubuntu

**Comment by nrdlngr** [comment]
@benhg, @errordeveloper, what was the configuration issue? It seems like others are having the same problem.

**Comment by nrdlngr** [comment]
Does anyone know what causes the "Post https://sts.amazonaws.com/: net/http: TLS handshake timeout" error?

**Comment by benhg** [comment]
I believe it was a problem with my EC2 credentials

**Comment by qmilangowin** [comment]
I'm still getting this, macOS Catalina, latest version of Docker Desktop with Kubernetes installed.
It's very frustrating. Had to create my own docker image to be able to use eksctl with our clusters at work. 
Cannot use it natively from my mac.

**Comment by michaelbeaumont** [comment]
@qmilangowin Are you able to use AWS cli otherwise?

**Comment by qmilangowin** [comment]
@michaelbeaumont Yes, I can use it no problem otherwise. Just with eksctl I have this issue.

**Comment by HTMLGuyLLC** [comment]
Was working fine, now it doesn't and I get `An error occurred (InvalidClientTokenId) when calling the GetCallerIdentity operation: The security token included in the request is invalid.`. So frustrating after trying a million things to fix it.

**Comment by aniketkalamkar** [comment]
> Is it possible to reopen this issue? I can try to help provide any details necessary to reproduce and fix it.
> 
> Thanks.

I am getting this issue lot of time -
aniket@PIXACOREs-MBP ~ % eksctl get cluster --verbose 5                       
Error: checking AWS STS access – cannot get role ARN for current session: RequestError: send request failed
caused by: Post "https://sts.amazonaws.com/": net/http: TLS handshake timeout

**Comment by msolimans** [comment]
error is not clear enough to understand the issue, if you're missing `--region` param or have a wrong value specified for region or don't have default region configured in ~/.aws/credentials you would encounter the same meaningless error

**Comment by espuuks** [comment]
Issue "env" and see what AWS_* variables are applying to you at the moment. In order to enforce the default profile, you may need to unset these three variables, if you have recently used STS to assume some role:

unset AWS_SESSION_TOKEN
unset AWS_SECRET_ACCESS_KEY
unset AWS_ACCESS_KEY_ID

**Comment by almeida-matheus** [comment]
after updating macOS to version 12 Monterey I can't use aws-vault, every time I run it has a delay of a few seconds and the following output error:

`aws-vault exec` with AWS SSO:
```
aws-vault: error: exec: Failed to get credentials for profile-name: RequestError: send request failed
caused by: Post "https://oidc.us-east-1.amazonaws.com/client/register": net/http: TLS handshake timeout
```

`aws-vault rotate` with access key credentials:
```
aws-vault: error: rotate: RequestError: send request failedcaused by: Post "https://sts.amazonaws.com/": net/http: TLS handshake timeout
```

does anyone have any idea how to solve this?

**Comment by tracetechnical** [comment]
Still seeing this on Ubuntu

**Comment by piyumisudusinghe** [comment]
I got a similar issue in windows when creating a cluster in terminal using eksctl

Error: checking AWS STS access – cannot get role ARN for current session: operation error STS: GetCallerIdentity, https response error StatusCode: 403, RequestID: af1e7e9f-19fd-4e54-8873-1ee6a2e55a6c, api error SignatureDoesNotMatch: The request signature we calculated does not match the signature you provided. Check your AWS Secret Access Key and signing method. Consult the service documentation for details.

The root casue for my scenario was my default aws profile configured via AWS CLI got modified and secrect access key got changed.
After modifed the aws credentials file in the local machine I was able to resolve the issue
